internal/controller/secret.go

```
@@ -106,31 +110,39 @@ func updateEntitlementPullSecrets(secret []byte, ctx context.Context, full kuber
 			"kubernetes.io/dockerconfigjson",
 			nil,
 		)
-		if extraPullSecret != nil {
-			mergedSecret, err := utils.MergeDockerSecrets(ibmPullSecret, extraPullSecret)
-			if err == nil {
-				ibmPullSecret = mergedSecret
-			}
-		}
-		_, err = full.CoreV1().Secrets(destNamespace).Get(ctx, destSecretName, metav1.GetOptions{})
-		if err != nil {
-			if kerrors.IsNotFound(err) {
-				// Resource does not exist, create it
-				_, err := full.CoreV1().Secrets(destNamespace).Create(context.TODO(), ibmPullSecret, metav1.CreateOptions{})
-				if err != nil {
-					return err
-				}
-				log.Log.Info(fmt.Sprintf("Created Secret %s in ns %s", destSecretName, destNamespace))
-				continue
-			}
-			return err
-		}
-		// The destination secret already exists so we upate it and return an error if they were different so the reconcile loop can restart
-		_, err = full.CoreV1().Secrets(destNamespace).Update(context.TODO(), ibmPullSecret, metav1.UpdateOptions{})
+		mergedSecret, err := utils.MergeDockerSecrets(ibmPullSecret, extraPullSecret)
 		if err == nil {
-			log.Log.Info(fmt.Sprintf("Updated Secret %s in ns %s", destSecretName, destNamespace))
-			continue
+			ibmPullSecret = mergedSecret
+		}
```

Looks like we don't log this error in any way, we just fail silently. Is that expected here?